src/compiler.js

```diff
@@ -94,6 +96,14 @@ Compiler.prototype.compileArgument = function (element) {
                 format: new Intl.NumberFormat(locales, options).format
             };
 
+        case 'shortNumberFormat':
+            options = formats.number;
```

Comment:
@jasonmit my last question is - should we be using the User Defined Formats here? 

```
var msg = new IntlMessageFormat('My short number: {numPeople, shortNumber, options}', 'en-US', {
    number: {
        options: {
            significantDigits   : 1,
            financialFormat: true
        }
    }
});
```
or is it better/simpler as what currently exists in this PR (does this even work in ember-intl?):

```
var msg = new IntlMessageFormat('My short number: {numPeople, shortNumber}', 'en-US', {
    number: {
          significantDigits   : 1,
          financialFormat: true
    }
});
```

Reply:
> or is it better/simpler as what currently exists in this PR (does this even work in ember-intl?):

The first example allows you to pass options to each formatting segment and the latrter appears to be universal for the whole translation.

I think if we're going to support one of these it should be the first one where we can pass the format hash key.  This way it allows a translation to contain different formatting options for each `shortNumber`.

```js
var msg = new IntlMessageFormat('My short number: {value, shortNumber, fin}', 'en-US', {
    shortNumber: { // <- we should not use number either
        fin: {
           significantDigits   : 1,
           financialFormat: true
       }
    }
});
```

Reply:
> @jasonmit What do you think about this? 30a1159

Looks good but can we remove the need to provide `options`?  Something like this would align well with the way we define other formatter options:

```
const formats = {
  shortNumber: {
    minimumFractionDigits: 2
  }
};
```